ui/components/multichain/address-list-item/address-list-item.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { ReactChild } from 'react';
+
+import { useSelector } from 'react-redux';
+import {
+  AvatarAccount,
+  Box,
+  AvatarAccountVariant,
+  Text,
+  AvatarAccountSize,
+} from '../../component-library';
+import {
+  TextAlign,
+  TextVariant,
+  FlexDirection,
+  BorderColor,
+  Display,
+  BlockSize,
+  BackgroundColor,
+  TextColor,
+} from '../../../helpers/constants/design-system';
+import { getUseBlockie } from '../../../selectors';
+
+interface AddressListItemProps {
+  label: string | ReactChild;
+  address: string;
+  onClick: () => void;
+}
+
+export const AddressListItem = ({
+  label,
+  address,
+  onClick,
+}: AddressListItemProps) => {
+  const useBlockie = useSelector(getUseBlockie);
+
+  return (
+    <Box
+      display={Display.Flex}
+      padding={4}
+      as="button"
+      onClick={(e: React.MouseEvent) => {
+        e.stopPropagation();
+        onClick();
+      }}
+      width={BlockSize.Full}
+      backgroundColor={BackgroundColor.transparent}
+      className="address-list-item"
+    >
+      <AvatarAccount
+        borderColor={BorderColor.transparent}
+        size={AvatarAccountSize.Sm}
+        address={address}
+        variant={
+          useBlockie
+            ? AvatarAccountVariant.Blockies
+            : AvatarAccountVariant.Jazzicon
+        }
+        marginInlineEnd={2}
+      />
+      <Box
+        display={Display.Flex}
+        flexDirection={FlexDirection.Column}
+        style={{ overflow: 'hidden' }}
+      >
+        <Text
+          variant={TextVariant.bodyMdMedium}
+          padding={0}
+          width={BlockSize.Full}
+          textAlign={TextAlign.Left}
+          ellipsis
+          data-testid="address-list-item-label"
+        >
+          {label}
+        </Text>
+        <Text
+          variant={TextVariant.bodySm}
+          color={TextColor.textAlternative}
+          ellipsis
+          data-testid="address-list-item-address"
+        >
+          {address}
+        </Text>
```

Comment:
I think we should have address shortening here rather than ellipsis? 
![Screenshot 2023-11-08 at 10 35 46 PM](https://github.com/MetaMask/metamask-extension/assets/39872794/cd6472cd-d266-4b21-a76c-db6c710a2d18)

Reply:
I wanted to show the full address, since it's an external account, but prod MetaMask shortens it so I think we can (1) shorten it and (2) provide a tooltip for the full address.